src/dune_rules/merlin/merlin.ml

```diff
@@ -247,35 +272,43 @@ module Processed = struct
     (* We only match the first part of the filename : foo.ml -> foo foo.cppo.ml
        -> foo *)
     let open Option.O in
-    let+ { module_; opens } =
+    let+ { module_; opens; reader } =
       let find file =
-        let file_without_ext = remove_extension file in
-        Path.Build.Map.find per_module_config file_without_ext
+        match Path.Build.Map.find per_module_config file with
+        | Some _ as s -> s
+        | None -> Path.Build.Map.find per_module_config (strip_pp_extensions file)
```

Comment:
Why do you need this fallbacking mechanism ?

Reply:
Modules with pp extensions `module_name.xxx.xx.ext` can be also valid modules residing in source directories (e.g. extra extension were not added by a preprocessor but present in source files). In this case it makes sense to try to resolve using non normalised filename first, I think.

Reply:
Do you have a test illustrating that ? In any case, I would like to take some time next week to pull your changes and understand better the relationship between source file / module / merlin config. It's a set of conventions that would benefit a lot from being correctly defined and documented.

The PR looks good otherwise 👍

Reply:
No, don't have a test case...

1. I'm actually wondering if we should remove any normalisation now that we have per filename config (rather per module) and such config is built by looking at source modules (so we see "real" source files).

1. Another thing to check (add test) is we still support promoted files.

For reference the behaviour of stripping the extension was added in c36df0b73ffa98f893788b25ce68139232299e41

Reply:
> Modules with pp extensions module_name.xxx.xx.ext can be also valid modules residing in source directories (e.g. extra extension were not added by a preprocessor but present in source files). In this case it makes sense to try to resolve using non normalised filename first, I think.

Added a test for that in my commit (at the end of the file), and it doesn't work right now.

> Another thing to check (add test) is we still support promoted files.
That's a good idea, can you do it ? (you can reuse the `merlin_config.sh` script in my new test to easily query the configuration.